dotnet/src/InternalUtilities/planning/PlanInstrumentation.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+/// <summary>Surrounds the invocation of a planner with logging and metrics.</summary>
+internal static partial class PlanInstrumentation
```

Comment:
I'd suggest just merging this with the PlannerInstrumentation class.

Reply:
Done